Can't install without ghc in PATH

The `cabal install Cabal-2.4.1.0` step in the makefile requires `ghc` in PATH, which isn't really the default for stack installations. Is there a way around this?

Sure. I'm running `make hie-8.2.2` and getting 

```
Warning: The update command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-update command
or the legacy v1-update alias as new-style projects will become the default in
the next version of cabal-install. Please file a bug if you cannot replicate a
working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Downloading the latest package list from hackage.haskell.org
To revert to previous state run:
    cabal update --index-state='2018-11-30T12:32:34Z'
cabal -- install Cabal-2.4.1.0
Warning: The install command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-install
command or the legacy v1-install alias as new-style projects will become the
default in the next version of cabal-install. Please file a bug if you cannot
replicate a working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

cabal.exe: The program 'ghc' version >=7.0.1 is required but it could not be
found.

make: *** [Makefile:95: cabal] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,13 @@ build: hie-8.2.1 hie-8.2.2 \
 build-all: build build-docs
 .PHONY: build-all
 
+GHC := $(shell stack path --compiler-exe)
+
+ghc:
+	$(GHC) --version
+	@echo GHC
+.PHONY: ghc
+
 # ------------------------------------------------------
 
 ## Builds hie for GHC version 8.2.1 only
@@ -91,8 +98,9 @@ submodules:
 ## NOTE 2: this is temporary, will go away once the new cabal-helper lands.
 ## NOTE 3: This is needed for stack only projects too
 cabal:
-	cabal update
-	cabal install Cabal-2.4.1.0
+	stack install cabal-install
+	cabal v1-update
+	cabal v1-install Cabal-2.4.1.0 --with-compiler=$(GHC)
 .PHONY: cabal
 
 # ------------------------------------------------------

